### ISSUE TITLE
Cannot compile Godot due error in SConstruct with SCons 4.0.1

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#91049]
### Tested versions

Works - 7316918a0f546f9fa406fd65b9a70161169426a8
Not works - 7abe0c6014022874378cb64a11b26b0f0f178324

### System information

Ubuntu 22.04 CI

### Issue description

` scons -j2 dev_build=yes target=editor use_asan=yes use_ubsan=yes linker=gold`

cause to show on Github CI this error
```
 KeyError: 'platform':
  File "/home/runner/work/Qarminer/Qarminer/godot/SConstruct", line 287:
    selected_platform = env["platform"]
  File "/usr/lib/python3/dist-packages/SCons/Environment.py", line 405:
    return self._dict[key]
```

Probably regression from - https://github.com/godotengine/godot/pull/90857

CI job - https://github.com/qarmin/Qarminer/actions/runs/8793092362

### Steps to reproduce

Above

### Minimal reproduction project (MRP)

N/A

**Comment by akien-mga** [comment]
CC @Repiteo

**Comment by akien-mga** [comment]
I can't reproduce the issue with SCons 4.7.0, but @qarmin's CI uses SCons 4.0.1.

I confirmed the bug with SCons 4.0.1 and 4.2.0, it seems like the syntax used in #90857 was added in SCons 4.3.0.

We currently still support SCons 3.0.0 as minimum version, to be able to support LTS Linux distros:
- Ubuntu 20.04 LTS has SCons 3.1.2 and Python 3.8.2
- Ubuntu 22.04 LTS has SCons 4.0.1 and Python 3.10.6

Maybe it's time to bite the bullet and use scons-local... But after removing their docbook sources it's still 68k extra lines of code that we'd vendor, and we'd run into issues where users don't have all the dependencies SCons needs that pip would have installed automatically.

The other option is to drop support for old Ubuntu SCons packages, and force those users to install SCons via pip. But we should test that pip is actually functional on these old distros, in my experience package managers like this tend to break after some time when the repos become centered around the current versions of the language, but your system version is old (like Python 3.8.2 on Ubuntu 20.04).
